src/Databases/TablesLoader.cpp

```diff
@@ -106,16 +108,44 @@ LoadTaskPtrs TablesLoader::startupTablesAsync(LoadJobSet startup_after)
 {
     LoadTaskPtrs result;
     std::unordered_map<String, LoadTaskPtrs> startup_database; /// database name -> all its tables startup tasks
+    TablesDependencyGraph all_startup_dependencies("AllStartupMvDependencies");
+    all_startup_dependencies = all_loading_dependencies;
+    // all_startup_dependencies.mergeWith(mv_to_dependencies);
 
-    for (const auto & table_id : all_loading_dependencies.getTables())
+    for (const auto & table_id : mv_to_dependencies.getTables())
     {
-        // Make startup table task
+        auto storage_id_vector = mv_to_dependencies.getDependencies(table_id);
+        for (const auto & storage_id : storage_id_vector)
+            all_startup_dependencies.addDependency(storage_id, table_id);
+    }
+    for (const auto & table_id : mv_from_dependencies.getTables())
+    {
+        auto storage_id_vector = mv_from_dependencies.getDependencies(table_id);
+        for (const auto & storage_id : storage_id_vector)
+            all_startup_dependencies.addDependency(table_id, storage_id);
+    }
```

Comment:
Please add a few comments describing why we need mv dependencies and how they affect the graph of AsyncLoader jobs. Startup job dependencies are becoming non-obvious.